File: src/my_cpp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

using namespace std::chrono_literals;

class RobotNewsStationNode : public rclcpp::Node
{   
    public:
        RobotNewsStationNode() : Node("robot_news_station_node")
        {
            RCLCPP_INFO(this->get_logger(), "Robot news station has been started.");
            publisher_ = this->create_publisher<example_interfaces::msg::String>("robot_news", 10);
            timer_ = this->create_wall_timer(
                500ms,
                std::bind(&RobotNewsStationNode::publish_news, this));
        }

    private:
        rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
        std::string robot_name_ = "Robot C30";
        rclcpp::TimerBase::SharedPtr timer_;
        int counter_ = 0;

        void publish_news()
        {
            auto  msg = example_interfaces::msg::String();
            msg.data = std::string("Hello, this is  ")
                + robot_name_ 
                + std::string(" from the robot news station ") 
                + std::to_string(counter_)
                + std::string(" !");
            publisher_->publish(msg);
            counter_++;
        }

};

int main(int argc, char ** argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotNewsStationNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

/**
Build and run the node:
$ colcon build --packages-select my_cpp_pkg
$ ros2 run my_cpp_pkg robot_news_station
*/

File: src/my_py_pkg/my_py_pkg/robot_news_station.py
#! /usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String


class RobotNewsStationNode(Node):
    def __init__(self):
        super().__init__("robot_news_station")
        self.robot_name_ = "C30"
        
        self.declare_parameter("robot_name", "C110")
        self.robot_name_ = self.get_parameter("robot_name").value
        
        self.publisher_ = self.create_publisher(String, "robot_news", 10)
        self.timer_ = self.create_timer(0.5, self.publish_news)
        self.get_logger().info("Robot News Station has been started.")        
        self.count_ = 0

    def publish_news(self):
        msg = String()
        msg.data = f"Hi, this is {self.robot_name_} from the Robot News Station {self.count_}!"
        self.publisher_.publish(msg)
        self.count_ += 1


def main(args=None):
    rclpy.init(args=args)
    node = RobotNewsStationNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()

"""
If you create the workspace the incorret way, you can rebuild it but you need delete built, install and log folder:

$ rm -rf build/ install/ log/

Build my package:

$ colcon build --packages-select my_py_pkg

run robot_news_station:

$ ros2 run my_py_pkg robot_news_station
"""